File: KOKOPangUser/src/main/java/com/koko/kokopang/user/model/Friendship.java
package com.koko.kokopang.user.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@Entity
@ToString
public class Friendship {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name="userId",referencedColumnName = "id")
    private User user;

    private int friendId;
    private int friendRating;
    private Boolean isWaiting = true; // 수락대기중인지 아닌지
    private Boolean isFrom = true; // 내가 보낸건지(true) 상대방이 보낸건지(false)
}

File: unity/Assets/Scripts/Data/Friendship.cs
using UnityEngine;

[System.Serializable]
public class Friendship
{
    [SerializeField] int userId;
    [SerializeField] int friendId;
    [SerializeField] string friendName;
    [SerializeField] int friendRating;
    [SerializeField] string friendProfileImg;
    [SerializeField] bool isWaiting;
    [SerializeField] bool isFrom;

    public int UserId
    {
        get
        {
            return userId;
        }
        set
        {
            userId = value;
        }
    }

    public int FriendId
    {
        get
        {
            return friendId;
        }
        set
        {
            friendId = value;
        }
    }

    public string FriendName
    {
        get
        {
            return friendName;
        }
        set
        {
            friendName = value;
        }
    }

    public int FriendRating
    {
        get
        {
            return friendRating;
        }
        set
        {
            friendRating = value;
        }
    }

    public string FriendProfileImg
    {
        get
        {
            return friendProfileImg;
        }
        set
        {
            friendProfileImg = value;
        }
    }

    public bool IsWaiting
    {
        get
        {
            return isWaiting;
        }
        set
        {
            isWaiting = value;
        }
    }

    public bool IsFrom
    {
        get
        {
            return isFrom;
        }
        set
        {
            isFrom = value;
        }
    }
}
